advisor-ui/src/pages/VolunteerHours/VolunteerHours.jsx

```diff
@@ -0,0 +1,122 @@
+import React, { useEffect, useContext, useState } from "react";
+import Table from "@mui/material/Table";
+import TableBody from "@mui/material/TableBody";
+import TableCell from "@mui/material/TableCell";
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+import axios from "axios";
+import { UserContext } from "../../UserContext.js";
+import { IconButton } from "@mui/material";
+import DeleteIcon from "@mui/icons-material/Delete";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import { Link } from "react-router-dom";
+import "./VolunteerHours.css";
+
+export default function VolunteerHours() {
+  const [studentEvents, setStudentEvents] = useState([]);
+  const { user } = useContext(UserContext);
+
+  const fetchStudentEvents = async () => {
+    try {
+      const res = await axios.get(
+        `http://localhost:5000/api/student-event/${user.id}`,
+        { params: { studentId: user.id } }
+      );
+      const fetchedStudentEvents = res.data.studentEvents;
+      setStudentEvents(fetchedStudentEvents);
+    } catch (error) {
+      alert("Something went wrong!");
+    }
+  };
+
+  useEffect(() => {
+    fetchStudentEvents();
+  }, []);
```

Comment:
This is an antipattern; `useEffect` is for things you need the component to do after rendering. Passing in an empty array tells the component that it only needs to do something once and I'd really avoid it for pretty much anything except _maybe_ impression logging. 

For this case, you have one dependency - the student's ID - but it's abstracted into the `fetchStudentEvents` function. _But_, I don't think you need either of these functions - you should be able to have this component fetch the data and then render it directly:

```
export default function VolunteerHours() {
  const [studentEvents, setStudentEvents] = useState([]);
  const { user } = useContext(UserContext); 

  try {
    const res = await axios.get(
      `http://localhost:5000/api/student-event/${user.id}`,
      { params: { studentId: user.id } }
    );
    setStudentEvents(res.data.studentEvents);
  } catch (error) {
    alert("Something went wrong!");
  }
...
```